### PR TITLE
Test transcript

### PR DESCRIPTION
**Pull request by helias22**
## Pull Request Summary
Automation for creating dummy transcript.

### Change types

- [x] Non-breaking change
- [] Adding tests
- [x] Front end change

### Testing
- [ ] I have added testing to this PR 

## Questions
I'm not sure how to call generateFile in FileGenerator. 
The way I did it was temporarily calling it in a test file. It works but there should be a better way?

### CODE REVIEW

File: front-end/src/components/main-page/TranscriptUploadModal.jsx [hunk 1]
```diff
@@ -55,8 +55,10 @@ class TranscriptUploadModal extends React.Component {
       const parsedData = JSON.parse(data);
       if (parsedData.data){
         for (const intent of parsedData.data) {
-          if (intentIsMissingFields(intent)){
-            return false;
+          if (Object.keys(intent).length !== 0){
+            if (intentIsMissingFields(intent)){
+              return false;
```
Comment (WeissSa) [inline front-end/src/components/main-page/TranscriptUploadModal.jsx:60]: Would this not throw an error if we passed in an empty object?

File: back-end/helpers/test-helpers/transcriptFactory.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        // give the created root of that tree as the first input 
+        // Second input is more confusing. you should pass a list of lists 
+        // each inner list has the number of occurrence of and the child node. 
+        // the number of occurrence doesn't need to be exact. think of it kinda as RV :)) 
+        // you can costumize that number to get the persentages you want
+        const b6t = TranscriptFactory.generateTranscript(b6, [[1,ct]])
+        const b3t = TranscriptFactory.generateTranscript(b3, [[1,e]])
+        const at = TranscriptFactory.generateTranscript(a, [[2,b6t],[1,b3t]]) 
+    });
+});
```
Comment (WeissSa) [inline back-end/helpers/test-helpers/transcriptFactory.spec.js:22]: Im confused why this is a spec file

File: back-end/helpers/test-helpers/transcriptFactory.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    static generateSingleEntry(id, type, message, name = "", isLeaf = false) {
+        let entry = {};
+        if(type == "text"){
+            entry = {
+                "project_id":id,
+                "trace_type":"text",
+                "trace_payload":"{\"type\":\"text\",\"payload\":{\"slate\":{\"id\":\"" + id + 
+                "\",\"content\":\"[{\\\"children\\\":[{\\\"text\\\":\\\"" + message +
+                "\\\"}]}]\"},\"message\":\"[INSERT TEXT]\"}}"
```
Comment (WeissSa) [inline back-end/helpers/test-helpers/transcriptFactory.js:24]: Should the message say "[INSERT TEXT]" here?

File: back-end/helpers/test-helpers/transcriptFactory.spec.js (hunk 2 above)
Comment (helias22) [reply back-end/helpers/test-helpers/transcriptFactory.spec.js:22]: My way of debugging don't worry about it. I'll move it. I just knew I couldn't spend much time today on this so I just wanted to give anyone that wants the option to use it even before it's completed.

File: front-end/src/components/main-page/TranscriptUploadModal.jsx (hunk 1 above)
Comment (helias22) [reply front-end/src/components/main-page/TranscriptUploadModal.jsx:60]: It wouldn't and that what we want isn't it? we want to use empty objects to differentiate between conversations in transcript

**Review by WeissSa — APPROVED** [review]
lgtm

**Review by KayleeLChan — APPROVED** [review]
funky file additions for that transcript goodness from our backend goddess 💃 💃 💃

**Pull request merged**